### PR TITLE
As a user, I can sync from registries that use basic auth.

### PR DESCRIPTION
**Pull request by ipanova**
This PR enables sync from the registries that use basic auth or
token auth but repos are protected or private.

closes #4717
closes #4808
https://pulp.plan.io/issues/4717
https://pulp.plan.io/issues/4808

### CODE REVIEW

File: pulp_docker/app/downloaders.py [hunk 1]
```diff
@@ -107,24 +119,31 @@ def __init__(self, *args, **kwargs):
...
     @staticmethod
-    def auth_header(token):
+    def auth_header(token, basic_auth):
         """
-        Create an auth header that optionally includes a bearer token.
+        Create an auth header that optionally includes a bearer token or basic auth.
 
         Args:
-            token (str): Bearer token to use in header
+            auth (str): Bearer token or Basic auth to use in header
```
Comment (asmacdo) [inline pulp_docker/app/downloaders.py]: this should be token and basic_auth

File: pulp_docker/app/downloaders.py [hunk 2]
```diff
@@ -107,24 +119,31 @@ def __init__(self, *args, **kwargs):
...
 
         Returns:
-            dictionary: containing Authorization headers or {} if token is None.
+            dictionary: containing Authorization headers or {} if Authorizationis None.
```
Comment (asmacdo) [inline pulp_docker/app/downloaders.py]: authorization is

File: pulp_docker/app/downloaders.py (hunk 1 above)
Comment (ipanova) [reply pulp_docker/app/downloaders.py]: actually no, it should one or another. Registry can have whether no auth, token auth or basic auth.
tldr, in the 401 auth header you will see only basic ir bearer scheme.
